forum/src/main/java/forum/controllers/GroupController.java

```diff
@@ -0,0 +1,66 @@
+package forum.controllers;
+
+import forum.entities.Group;
+import forum.entities.Topic;
+import forum.services.GroupService;
+import forum.services.TopicService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/groups")
+public class GroupController {
+
+    private final GroupService groupService;
+    private final TopicService topicService;
+
+    @Autowired
+    public GroupController(GroupService groupService, TopicService topicService) {
+        this.groupService = groupService;
+        this.topicService = topicService;
+    }
+
+    @GetMapping
+    public @ResponseBody
+    List<Group> getGroups() {
+        return groupService.findAll();
+    }
+
+    @GetMapping("{groupId}")
```

Comment:
It's better to add slash to path. It will work without it, but let's use more obvious approach.